Provider/Pool.php

```diff
@@ -77,12 +84,27 @@ public function addProvider($name, MediaProviderInterface $instance)
     }
 
     /**
+     * @deprecated since version 3.0, to be removed in 4.0.
+     *
      * @param string                    $name
      * @param DownloadStrategyInterface $security
      */
     public function addDownloadSecurity($name, DownloadStrategyInterface $security)
     {
+        @trigger_error('The '.__METHOD__.' method is deprecated since version 3.0 and will be removed in 4.0.', E_USER_DEPRECATED);
+
         $this->downloadSecurities[$name] = $security;
```

Comment:
No, this method add a new entry to `$this->downloadSecurities`, the other to the old `$this->downloadStrategies`. This is needed to use the getters for both properties.